pages/en/lb4/Creating-components.md

```diff
@@ -307,3 +307,297 @@ class EmailComponent {
+export type Person = {
+  name: string;
+  hobby: string;
+};
+```
+
+### Defining our router and server
+Now that there is a controller to test with, the following code creates a `Router` and `Server`.
+This example uses [express](https://expressjs.com/) and the
+`@types/express` package.
+
+{% include code-caption.html content="/src/server/rpc-router.ts" %}
+```ts
+import {RPCServer} from './rpc-server';
+import * as express from 'express';
+import * as parser from 'body-parser';
+
+export class RPCRouter {
+  routing: express.Router;
+  constructor(private server: RPCServer) {
+    this.routing = express.Router();
+    const jsonParser = parser.json();
+    this.routing.post('*', jsonParser, async (request, response) => {
+      console.log(JSON.stringify(request.body));
+      // Aliasing these values for clarity and brevity.
+      const ctrl = request.body.controller;
+      const method = request.body.method;
+      const input = request.body.input;
+      try {
+        const controller = await this.server.get(`controllers.${ctrl}`);
+        if (!controller) {
+          response.statusCode = 400;
+          response.send(`No controller was found with name "${ctrl}".`);
+          return;
+        }
+        if (!controller[method]) {
+          response.statusCode = 400;
+          response.send(
+            `No method was found on controller "${ctrl}" with name "${method}".`
+          );
+        }
+        response.send(await controller[method](input));
+      } catch (err) {
+        // You'll probably want more robust error handling in a real server!
+        console.error(err);
+        response.statusCode = 500;
+        response.send(err);
+      }
+    });
+    // Use our router!
+    this.server.expressServer.use(this.routing);
+  }
+}
+```
+
+In addition to the server, the example will include a small type definition for
+convenience.
+
+{% include code-caption.html content="/src/server/rpc-server.ts" %}
+```ts
+import {inject, Context} from '@loopback/context';
+import {Server, Application, CoreBindings} from '@loopback/core';
+import {RPCRouter} from './rpc-router';
+import * as express from 'express';
+import * as http from 'http';
+
+export class RPCServer extends Context implements Server {
+  _server: http.Server;
+  expressServer: express.Application;
+  router: RPCRouter;
+  constructor(
+    @inject(CoreBindings.APPLICATION_INSTANCE) public app?: Application,
+    @inject('rpcServer.config') public config?: RPCServerConfig
+  ) {
+    super(app);
+    this.config = config || {};
+  }
+
+  async start(): Promise<void> {
+    this.expressServer = express();
+    this.router = new RPCRouter(this);
+    // Currently, there is a limitation with core node libraries
+    // that prevents the easy promisification of some functions, like
+    // http.Server's listen method.
+    this._server = this.expressServer.listen(
+      (this.config && this.config.port) || 3000
+    );
+    return Promise.resolve();
```

Comment:
This line is not needed at all. When an `async` function does not return anything, the runtime will convert it into a promise resolving to `undefined` under the hood.

Please remove the other usages of `return Promise.resolve()` too.